pkg/apis/elasticsearch/v1/validations.go

```diff
@@ -207,29 +204,6 @@ func checkNodeSetNameUniqueness(es *Elasticsearch) field.ErrorList {
 	return errs
 }
 
-// pvcModification ensures no PVCs are changed, as volume claim templates are immutable in stateful sets
-func pvcModification(current, proposed *Elasticsearch) field.ErrorList {
-	var errs field.ErrorList
-	if current == nil || proposed == nil {
-		return errs
-	}
-	for i, node := range proposed.Spec.NodeSets {
-		currNode := getNode(node.Name, current)
-		if currNode == nil {
-			// this is a new sset, so there is nothing to check
-			continue
-		}
-
-		// ssets do not allow modifications to fields other than 'replicas', 'template', and 'updateStrategy'
-		// reflection isn't ideal, but okay here since the ES object does not have the status of the claims.
-		// Checking semantic equality here allows providing PVC storage size with different units (eg. 1Ti vs. 1024Gi).
-		if !apiequality.Semantic.DeepEqual(node.VolumeClaimTemplates, currNode.VolumeClaimTemplates) {
-			errs = append(errs, field.Invalid(field.NewPath("spec").Child("nodeSet").Index(i).Child("volumeClaimTemplates"), node.VolumeClaimTemplates, pvcImmutableMsg))
-		}
-	}
-	return errs
-}
```

Comment:
It would allow:
- exact same volumeClaimTemplates content
- increased storage requests only if the underlying storage class allows it

It would disallow everything else, for example:
- any other change in the volumeClaimTemplates
- storage requests decrease